test/python/test_sampler.py

```diff
@@ -0,0 +1,58 @@
+from __future__ import annotations
+
+import numpy as np
+import pytest
+from qiskit import ClassicalRegister, QuantumCircuit, QuantumRegister
+from qiskit.circuit import Parameter
+from qiskit.transpiler.passes import Unroller
+from qiskit.transpiler.passmanager import PassManager
+
+from mqt.ddsim.sampler import DDSIMBackendSampler
+
+
+@pytest.fixture()
+def sampler() -> DDSIMBackendSampler:
+    """The sampler fixture for the tests in this file."""
+    return DDSIMBackendSampler()
+
+
+@pytest.fixture()
+def circuit() -> QuantumCircuit:
+    """The circuit fixture for the tests in this file."""
+
+    theta_a = Parameter("theta_a")
+    theta_b = Parameter("theta_b")
+    qc = QuantumCircuit(name="test")
+    q = QuantumRegister(1, "q")
+    c = ClassicalRegister(1, "c")
+    qc.add_register(q)
+    qc.add_register(c)
+    qc.rx(theta_a, 0)
+    qc.rz(theta_b, 0)
+    qc.rx(theta_a, 0)
+    qc.barrier()
+    qc.measure(q, c)
+
+    return qc
+
+
+def test_ddsim_sampler(circuit: QuantumCircuit, sampler: DDSIMBackendSampler):
+    """Test DDSIM Sampler's functionality"""
+
+    job = sampler.run([circuit], [[np.pi / 2, np.pi]])
+    result = job.result()
+    counts = result.quasi_dists[0]
+
+    assert counts == {0: 1.0}
```

Comment:
Testing a couple more scenarios would be good here. Maybe have a look at the Qiskit test suite for their sampler and make use of some of the tests over there.

Reply:
Do you want me to include tests that involve counts with multiple outcomes? I remember last time you told me that it was better to keep single outcome tests with 100% probability to avoid using thresholds.

Reply:
Hm. Have you looked at the Qiskit test suite and what they are doing?
Typically I would want to avoid these kinds of tests that are prone to fail randomly.
However, they are almost necessary I guess to test more complex scenarios.

Reply:
I was looking at it. I added more complex tests for different cases based on what I found.
Several tests I found involved circuits with multiple outcomes, so I sticked to that metric. We can always adapt the tolerance if you want to avoid random fails.